modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -654,32 +674,45 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
     {
+        std::cout << "Use WebNN Pooling Layer's Implementation." << std::endl;
         Ptr<WebnnBackendNode> node = nodes[0].dynamicCast<WebnnBackendNode>();
         auto& webnnInpOperand = node->operand;
         auto& webnnGraphBuilder = node->net->builder;
+        Pool2dOptions options;
+        std::vector<int32_t> kernelSize(kernel_size.begin(), kernel_size.end());
+        std::vector<int32_t> Strides(strides.begin(), strides.end());
+        std::vector<int32_t> Padding;
+        if (padMode.empty()) {
+            Padding = {static_cast<int32_t>(pads_begin[0]),
+            static_cast<int32_t>(pads_end[0]),
+            static_cast<int32_t>(pads_begin[1]),
+            static_cast<int32_t>(pads_end[1])};
+        } else if (padMode == "VALID") {
+            Padding = {0, 0, 0, 0};
+        } else if (padMode == "SAME") {
+            options.autoPad = ml::AutoPad::SameUpper;            
+        }
+        // std::cout<<"ceilMode: "<<ceilMode<<std::endl;
+        // std::cout << "padMode: " << padMode << std::endl;
+        // std::cout << "type: " << type << std::endl;
+        // std::cout<<"kernelSize"<<kernelSize[0]<<" "<<kernelSize[1]<<std::endl;
+        // std::cout<<"Strides"<<Strides[0]<<" "<<Strides[1]<<std::endl;
+        // std::cout<<"Padding:"<<Padding[0]<<" "<<Padding[1]<<" "<<Padding[2]<<" "<<Padding[3]<<std::endl;
+        // std::cout<<"avePoolPaddedArea:"<<avePoolPaddedArea<<std::endl;
+        options.windowDimensions = kernelSize;
+        options.strides = Strides;
+        options.padding = Padding;
         if (type == MAX)
         {
-            Pool2dOptions options;
-            std::vector<int32_t> kernelSize(kernel_size.begin(), kernel_size.end());
-            std::vector<int32_t> Strides(strides.begin(), strides.end());
-            // std::vector<int32_t> Padding(pads_begin.begin(), pads_begin.end());
-            options.windowDimensions = kernelSize;
-            options.strides = Strides;
-            // options.padding = Padding;
             auto operand = webnnGraphBuilder.MaxPool2d(webnnInpOperand, options.AsPtr());
             return Ptr<BackendNode>(new WebnnBackendNode(operand));
         }
-        else
+        else if (type == AVE)
         {
-            Pool2dOptions options;
-            std::vector<int32_t> kernelSize(kernel_size.begin(), kernel_size.end());
-            std::vector<int32_t> Strides(strides.begin(), strides.end());
-            // std::vector<int32_t> Padding(pads_begin.begin(), pads_begin.end());
-            options.windowDimensions = kernelSize;
-            options.strides = Strides;
-            // options.padding = Padding;
             auto operand = webnnGraphBuilder.AveragePool2d(webnnInpOperand, options.AsPtr());
             return Ptr<BackendNode>(new WebnnBackendNode(operand));
+        } else {
+            CV_Error(Error::StsNotImplemented, "Unsupported pooling type");
```

Comment:
It would be good to log which type of pooling is not supported.

Reply:
Hi Ningxin, I noticed that this line of code won't run actually since if ```type != MAX && type != AVE```, the ```supportBackend()``` function will return false. The reason why I add this code here is to avoid the error that "the function may return void" during the compiling process. To report which type of pooling is not supported, a CV_LOG_WARNING can be added to the ```supportBackend()``` function.